app/lib/controllers/api.js

```diff
@@ -35,203 +110,155 @@ const dbApi = require('./dbApi')
-* @param {qry}- a comma separated list of disaster id's
+* @function /disasternumber/:qry
+* @param {string} :qry - a comma separated list of disaster id's
+* @example // returns disaster array that is limited by the specified disasters (in this case, the Wisconsin cities Madison and Monroe)
+*  get /disasternumber/DR-4311-UT,FM-5130-UT,FM-5182-WA
 */
 router.get('/disasternumber/:qry', function (req, res) {
+  const authorizedDisasters = _.get(req, 'query.disasters')
   const disasterNbrs = req.params.qry.toUpperCase().split(',')
   var filter = ''
   for (var arg in disasterNbrs) {
     var qryParts = disasterNbrs[arg].split('-')
     if (arg > 0) filter += ' or '
     filter += `( disasterType eq '${qryParts[0]}' and disasterNumber eq ${qryParts[1]} and state eq '${qryParts[2]}' )`
   }
-  const qs = querystring.stringify({
-    $filter: `${filter}`
+  fema.getDisasters({
+    filter: `${filter}`,
+    authorizedDisasters
+  }, (err, data) => {
+    if (err) return res.send(503, {status: err.status, message: err.message})
+    res.json(data)
   })
-  const url = `https://www.fema.gov/api/open/v1/DisasterDeclarationsSummaries?${qs}`
-  console.log(url)
-  request(url, function (err, response, body) {
-    // TODO handle error for cases when FEMA API responds with body starting wth '<!DOCTYPE html>' +bug id:2 gh:9
-    var data
-    try {
-      data = JSON.parse(body, (key, value) => {
-        return key === 'declarationDate' ? moment(value).format('MMMM DD, YYYY') : value
-      })
-    } catch (e) {
-      return res.send(503, {status: response.statusCode, message: e.message})
-    }
-    var rolledUpData = rollUpData(data)
-    res.json(rolledUpData)
+})
+
+/**
+* router.get('/export/:fileNamePart') <br/>
+*  Generates a CSV file with all the columns from the database<br/>
+* @function /export/:fileNamePart
+* @param {string} :fileNamePart - a string that will be appended to 'hud-fema-data-' to form the download filename
+* @param {string} disasters - a comma separated list of disaster id's
+* @example // returns CSV for disasters DR-4311-UT,FM-5130-UT, and FM-5182-WA with download filename set to hud-fema-data-MY_FILE_NAME
+*  get /export/MY_FILE_NAME?disasters=DR-4311-UT,FM-5130-UT,FM-5182-WA
+*/
+router.get('/export/:fileNamePart', function (req, res) {
+  var fileName = `hud-fema-data-${req.params.fileNamePart}`
+  var disasterNumbers = _.get(req, 'query.disasters').split(',')
+  if (!disasterNumbers || disasterNumbers[0].length === 0) return res.status(406).send('No disaster numbers sent. Not Acceptable.')
+  var states = _.uniq(_.map(disasterNumbers, d => d.split('-')[2]))
+  var numbers = _.uniq(_.map(disasterNumbers, d => d.split('-')[1]))
+  if (states[0] === undefined || numbers[0] === undefined) return res.status(406).send('Invalid disaster numbers sent. Not Acceptable.')
+  var query = `states=${states.join(',')}&disasters=${numbers.join(',')}`
+  const uri = `http://${req.headers.host}/api/applicants/export?${query}`
```

Comment:
I'm nervous about hardcoding `http://`. Not a reason to reject the PR, but something to call up.

Reply:
Good catch @waldoj.  @davehannon there should be a req.protocol